sample_2.py

```diff
@@ -0,0 +1,17 @@
+class Solution:
+   def fib(self, N: int) -> int:
+       if N <= 1:
```

Comment:
line 3-4 is incorrect as per the logic
for any negative value it will return same value like if you pass -10 it will return -10
 you can correct it to 

if N <= 0:
    return 0
elif N == 1 :
     return 1

before this you can even add stamenet to check whether number passed is of int type using below statement
isinstance(N, int)

Reply:
Added error handling cases for `<0` and `==1` cases.